src/blockchain/wallet.ts

```diff
@@ -87,9 +94,78 @@ const listReleaseTransactionByWalletAddress = async (
   });
 };
 
+// list all Locks added by wallet adress
+const listValidLocksByWalletAddress = async (
+  walletAddress: string
+): Promise<Event[]>=> {
+  const provider = getProvider();
+
+  const signer = provider.getSigner();
+  const p2pContract = new ethers.Contract(getP2PixAddress(), p2pix.abi, signer);
+
+  const filterAddedLocks = p2pContract.filters.LockAdded([walletAddress]);
+  const eventsAddedLocks = await p2pContract.queryFilter(filterAddedLocks);
+
+  return eventsAddedLocks;
+};
+
+// list all Locks released by wallet adress
+const listReleasedLocksByWalletAddress = async (
+  walletAddress: string
+): Promise<Event[]> => {
+  const provider = getProvider();
+
+  const signer = provider.getSigner();
+  const p2pContract = new ethers.Contract(getP2PixAddress(), p2pix.abi, signer);
+
+  const filterReleasedLocks = p2pContract.filters.LockReleased([walletAddress]);
+  const eventsReleasedLocks = await p2pContract.queryFilter(
+    filterReleasedLocks
+  );
+  return eventsReleasedLocks;
+};
+
+// check locks added but didnt released
+const checkUnreleasedLocks = async (
+  walletAddress: string
+): Promise<any> => {
+  const addedLocks = await listValidLocksByWalletAddress(walletAddress);
+  const releasedLocks = await listReleasedLocksByWalletAddress(walletAddress);
+
+  if (addedLocks.length === 0) {
+    return false;
+  }
+
+  if (addedLocks.length === releasedLocks.length) {
+    return false;
+  };
+
+  const lock = addedLocks.find((addedLock) =>
+    releasedLocks.some(
+      (releasedLock) => releasedLock?.args?.lockId == addedLock?.args?.lockID
+    )
+  );
+
+  const provider = getProvider();
+
+  const signer = provider.getSigner();
+  const p2pContract = new ethers.Contract(getP2PixAddress(), p2pix.abi, signer);
```

Comment:
You can reduce this code by adding getContract method from provider file, instead of adding provider, signer and calling Contract.

Reply:
Fixed